StyleCop.Analyzers/StyleCop.Analyzers.CodeGeneration/SyntaxLightupGenerator.cs

```diff
@@ -30,22 +30,70 @@ private enum NodeKind
 
         public void Initialize(IncrementalGeneratorInitializationContext context)
         {
-            var compilation = context.CompilationProvider;
+            var referencedAssemblies = context.CompilationProvider.SelectMany(
```

Comment:
@chsienki this is `IncrementalValuesProvider<IAssemblySymbol>`, which should be efficient provided the `IAssemblySymbol` for a metadata reference is carried forward from one compilation to the next. Do you know if this is true for the IDE and/or command line?

Reply:
Yes, they should be carried forward. There is also `context.MetadataReferences` which provides the same thing. You don't get the `IAssemblySymbol` directly, but if you new up a compilation with the reference you can retrieve it, and the input will then never change unless the user adjusts references.

Reply:
I'll switch over to `context.MetadataReferences` after 4.2.0 is released (this is the version that corrects the signature for this property).